pkg/controller/vitesscluster/reconcile_topo.go

```diff
@@ -190,6 +223,17 @@ func (r *ReconcileVitessCluster) reconcileKeyspaceTopology(ctx context.Context,
 		desiredKeyspaces[keyspace.Name] = keyspace
 	}
 
+	if *vt.Spec.TopologyReconciliation.PruneKeyspaces == true {
```

Comment:
It looks like we only use `desiredKeyspaces` inside `pruneKeyspaces` so we should move it there.

Reply:
Done https://github.com/planetscale/vitess-operator/pull/26/commits/94069f9c3becd597518b1d7f7407b37f51282dc2.